### DISCUSSION TITLE
Splitting `tonic_build` code generation into SSR/CSR modules?

### DISCUSSION THREAD

**Discussion by lcmgh** [hyperium/tonic#2022] [Q&A]
The default setup as seen here https://github.com/hyperium/tonic/blob/master/tonic-build/README.md creates one large file for the actual grpc Client which includes structs for the request and response types.

I would like to keep my grpc client on the server side but use the response structs on the client-side-rendered web application. Has anyone found a method for doing so? I really don't like to manually duplicate the response structs. 

I found that when using 

```rust
        .build_transport(false)
        .build_client(false)
        .build_server(false)
``` 

only entities are created. So what I actually need is to config to define another file in which these entities are put. Then I could annotate them with a feature flag etc.

**Comment by rtritto** [comment]
@lcmgh this is a Feature Request and it should be opened as an issue, please can you do?
